internal/ethapi/simulate.go

```diff
@@ -73,6 +73,19 @@ func (r *simCallResult) MarshalJSON() ([]byte, error) {
 	return json.Marshal((*callResultAlias)(r))
 }
 
+// simBlockResult is the result of a simulated block.
+type simBlockResult struct {
+	sim   *simulator
```

Comment:
Feels weird to me to include a pointer to the simulator when you really only need the config + `fullTx`. But even still, I don't see why you need to return these at all. `execute` is a method on `sim` so the caller will always have `sim` around to do the RPCMarshalBlock correctly.

Reply:
I removed sim from simBlockResult and replaced it with fullTx + chainConfig.

You're right that this is not needed for the fix. It was a refactoring that made the fix nicer (but also generally the code I think). I disliked returning a map[string]interface{} and preferred to return a strict type. And moving the encoding logic from processBlock to its own method felt like a good change.